### PR TITLE
Environment config

### PR DESCRIPTION
**Pull request by dividedby0**
Added parameter envConfig for environment config which overrides every other config except args

### CODE REVIEW

File: packages/testring-typings/config.d.ts [hunk 1]
```diff
@@ -14,5 +14,6 @@ export interface IConfig {
     tests: string,
     httpThrottle: number,
     loggerLevel: number,
+    envConfig: string | void,
```
Comment (dividedby0) [inline packages/testring-typings/config.d.ts]: Unsure if this is right type for optional params

File: packages/testring-cli-config/src/arguments-parser.ts [hunk 2]
```diff
@@ -68,6 +68,12 @@ createField('loggerLevel', {
     type: 'number'
 });
 
+createField('envConfig', {
+    describe: 'path to environment config which overrides main config',
+    type: 'string',
+    default: undefined,
```
Comment (johnthecat) [inline packages/testring-cli-config/src/arguments-parser.ts]: by default it's already undefined, remove plz

File: packages/testring-cli-config/src/config-file-reader.ts [hunk 3]
```diff
@@ -13,25 +13,25 @@ const findFile = (configPath) => {
...
 
-const readJSConfig = async (userConfig: IConfig): Promise<IConfig | null> => {
+const readJSConfig = async (configPath: string): Promise<IConfig | null> => {
     try {
-        const configFile = require(userConfig.config);
+        const configFile = require(configPath);
 
         if (typeof configFile === 'function') {
-            return await configFile(userConfig);
+            return await configFile(configPath);
```
Comment (johnthecat) [inline packages/testring-cli-config/src/config-file-reader.ts]: you should pass here full config, add it as second argument for readJSConfig

File: packages/testring-typings/config.d.ts (hunk 1 above)
Comment (johnthecat) [reply packages/testring-typings/config.d.ts]: use `envConfig?: string`

**Pull request merged**